lib/base-controller.js

```diff
@@ -49,16 +50,17 @@ BaseController.prototype.locals = function controllerLocals(req, res) {
 };
 
 BaseController.prototype.getValues = function getValues(req, res, callback) {
-  // clear the session if there's no next step or we request to clear the session
-  if ((_.isUndefined(this.options.next) && this.options.clearSession !== false) || this.options.clearSession === true) {
-    req.sessionModel.reset();
-  }
-
   if (_.isEmpty(this.getErrors(req, res))) {
     this.referrer = req.header('Referer');
   }
 
   Controller.prototype.getValues.call(this, req, res, callback);
+
+  // clear the session if there's no next step or we request to clear the session
+  if ((_.isUndefined(this.options.next) && this.options.clearSession !== false) || this.options.clearSession === true) {
```

Comment:
because the the parent#getValues executes the callback, will this line of code ever be reached, assuming the callback calls res.render or summin't.

It could be better to pass a callback into the parent that executes our code first.

```js
Controller.prototype.getValues.call(this, req, res, function (err, values) {
    // clear the session if there's no next step or we request to clear the session
    if ((_.isUndefined(this.options.next) && this.options.clearSession !== false) || this.options.clearSession === true) {
        logger.info('Clearing session at endpoint', req.url);
        req.sessionModel.reset();
    }
    callback(err, values)
});
````